pkg/api/validation/validation.go

```diff
@@ -2083,9 +2155,51 @@ func ValidateReadOnlyPersistentDisks(volumes []api.Volume, fldPath *field.Path)
 	return allErrs
 }
 
+// validateTaints tests if given taints have valid data.
+func validateTaints(taints []api.Taint, fldPath *field.Path) field.ErrorList {
+	allErrors := field.ErrorList{}
+	for i, currTaint := range taints {
+		idxPath := fldPath.Index(i)
+		// validate the taint key
+		allErrors = append(allErrors, unversionedvalidation.ValidateLabelName(currTaint.Key, idxPath.Child("key"))...)
+		// validate the taint value
+		if !validation.IsValidLabelValue(currTaint.Value) {
```

Comment:
Does IsValidLabelValue() allow empty string? (The regexp uses ? so I think answer is yes?) If so, this accepts empty string as a valid label value. (And probably everything else that uses labels is doing the same.) We should probably file an issue about that, because I don't think we want empty string to be a valid label value. (But I would not try to fix it here.)